ebean-api/src/main/java/io/ebean/bean/InterceptReadOnly.java

```diff
@@ -1,39 +1,35 @@
 package io.ebean.bean;
 
+import io.ebean.UnloadedPropertyException;
+import io.ebean.UnmodifiableEntityException;
 import io.ebean.ValuePair;
 
-import java.util.Collections;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 /**
  * EntityBeanIntercept optimised for read only use.
  * <p>
  * For the read only use this intercept doesn't need to hold any state that is normally
- * required for updates such as per property changed, loaded, dirty state, original values
+ * required for updates such as per property changed, dirty state, original values
  * bean state etc.
  */
-public class InterceptReadOnly implements EntityBeanIntercept {
+public final class InterceptReadOnly extends InterceptBase {
 
-  private final EntityBean owner;
+  private final boolean[] flags;
```

Comment:
Maybe rename to "loaded" or "loadedFlags" .

For beans with <=64 properties, we even may introduce an memory optimizes intercept, that packs the loaded flags into a 64bit long value

Reply:
Renamed to "loaded", that will do for now.